### PR TITLE
Extracted pr_extract_char_follow_type into its own source files

### PR DESCRIPTION
**Pull request by bluedreamer**

### CODE REVIEW

**Review by Seifert69 — COMMENTED** [review]
Amazing, thanks :-)

File: vme/src/essential.h [hunk 1]
```diff
@@ -118,6 +118,7 @@ using ubit1 = bool;      /* Boolean */
 
 #define FREE(p)                                                                                                                            \
     {                                                                                                                                      \
+        /* Make sure FREE isn't being used to free class items created with EMPLACE */                                                     \
```
Comment (Seifert69) [inline vme/src/essential.h:121]: I was wondering if you made a unit_test for this code? It looks complicated, not sure I'd know how to test it if I had to :-)

**Review by Seifert69 — APPROVED** [review]

File: vme/src/essential.h (hunk 1 above)
Comment (bluedreamer) [reply vme/src/essential.h:121]: Don't really have to, its a compile time check - it won't compile if you pass in a class pointer :-)

But hopefully this template voodoo will go away as things get cleaned up

**Comment by Seifert69** [comment]
Thanks for the explanation :-)))

**Pull request merged**